### ISSUE TITLE
Cannot use JS reserved words as names of targeted hooked methods

### ISSUE THREAD

**Issue by 0xABD** [frida/frida#129]
In Frida 7.0.10, when writing hooks for e.g. Dalvik we cannot use Javacript's reserved words as targeted hooked methods, such as e.g. `delete` (the use case here is to hook `java.io.File.delete()`)
Attempting to execute the following script

```
    var hook01 = Dalvik.use("java.io.File");
    hook01.delete.overload().implementation = function() {
        send("File.delete() hooked");
        return Boolean(1);
    };
```

Will return the following error

```
SyntaxError: Unexpected token delete
    at makeMethod (gumjs-java.js:1248:38)
    at gumjs-java.js:978:28
    at Array.map (native)
    at makeMethodFromOverloads (gumjs-java.js:941:43)
    at gumjs-java.js:919:45
    at VM.perform (gumjs-java.js:2088:17)
    at File.Object.defineProperty.get (gumjs-java.js:910:36)
    at script1.js:45:11
    at VM.perform (gumjs-java.js:2088:17)
    at Array.forEach (native)
```

If we try to encode "delete" as inline unicode literals, like "\u0064elete", it does not work, too.

**Comment by oleavr** [comment]
This should work:

```
    var hook01 = Java.use("java.io.File");
    hook01['delete'].overload().implementation = function() {
        send("File.delete() hooked");
        return Boolean(1);
    };
```

(Btw, `Java` is the new name, `Dalvik` is a deprecated alias that will be removed at some point.)

**Comment by 0xABD** [comment]
Just tried that, returns the same error message 
```
SyntaxError: Unexpected token delete                            
    at makeMethod (gumjs-java.js:1248:38)                       
    at gumjs-java.js:978:28                                     
    at Array.map (native)                                       
    at makeMethodFromOverloads (gumjs-java.js:941:43)           
    at gumjs-java.js:919:45                                     
    at VM.perform (gumjs-java.js:2088:17)                       
    at File.Object.defineProperty.get (gumjs-java.js:910:36)    
    at script1.js:45:11                                         
    at VM.perform (gumjs-java.js:2088:17)                       
    at Array.forEach (native)                                   
```

**Comment by oleavr** [comment]
Ahh, I see the problem. Could you try:
- copy-pasting `gumjs-java.js` into a new file, e.g. *foo.js*
- changing the [module names](https://github.com/frida/frida-gum/blob/a4ef79b8a25101c2b9cc41faf918e027e542d662/bindings/gumjs/gumjs-java.js#L75-L89) to something unique to avoid colliding with Frida's builtin types
- adding your test-code to the bottom of the file, making sure to use the new module name that you defined in the previous step
- load that with `frida -U TargetProcessNameOrPid -l foo.js`

Then verify that it reproduces the problem, and attempt a fix:
- append an underscore to the name generated [here](https://github.com/frida/frida-gum/blob/a4ef79b8a25101c2b9cc41faf918e027e542d662/bindings/gumjs/gumjs-java.js#L1220)
- append an underscore to the name generated [here](https://github.com/frida/frida-gum/blob/a4ef79b8a25101c2b9cc41faf918e027e542d662/bindings/gumjs/gumjs-java.js#L1532)

**Comment by 0xABD** [comment]
Tested the fix successfully on Nexus 4 Android 4.4.2 KTU84P.
```
frida -U -l foobar.js -f "com.android.chrome"
```
[foobar.js.txt](https://github.com/frida/frida/files/191036/foobar.js.txt)
As a side note: I wonder whether the same issue applies to iOS? I don't have a test device, but I assume that there may be some reserved keywords conflicted too.

**Comment by oleavr** [comment]
Should be fixed in latest Frida 9.x.
